src/wasm/wasm-ir-builder.cpp

```diff
@@ -1323,7 +1323,7 @@ Result<> IRBuilder::makePop(Type type) {
       "pop instructions may only appear at the beginning of catch blocks"};
   }
   auto expectedType = scope.exprStack[0]->type;
-  if (type != expectedType) {
+  if (!Type::isSubType(expectedType, type)) {
```

Comment:
Shouldn't this be flipped? That is, the seen type should be a subtype of the expected type?

Reply:
No, in this case if I catch e.g. eqref, it would be valid and safe to overapproximate and have a `pop anyref`, but it would not be safe or valid to have `pop i31ref`, since the eqref I'm catching may not be an i31ref.